walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java

```diff
@@ -15,9 +15,144 @@
 package com.google.sps;
 
 import java.util.Collection;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Collections;
+import java.util.HashSet;
+import java.util.Arrays;
 
+// Meeting request: duration and attendes. 
+// Event: Time range, name of event, attendes of event. 
 public final class FindMeetingQuery {
+    
+  public ArrayList<TimeRange> gettingCalendar(
+                                                ArrayList<TimeRange> timeBlocks,
+                                                ArrayList<TimeRange> notAvailableTimes,
+                                                ArrayList<TimeRange> availableTimes,
+                                                long duration
+                                              ) {
+    TimeRange current = timeBlocks.get(0);
+    for (TimeRange time: timeBlocks) {
+
+      if (time.overlaps(current)) {
+        int start = current.start();
+          if (time.end() < current.end()) {
```

Comment:
This could be combined into a ternary if you want: `current = TimeRange.fromStartEnd(start, time.end() < current.end() ? current.end() : time.end(), false);`